trio/_tests/tools/test_gen_exports.py

```diff
@@ -3,13 +3,17 @@
 
 import pytest
 
-from trio._tools.gen_exports import (
-    File,
-    create_passthrough_args,
-    get_public_methods,
-    process,
-    run_linters,
-)
+try:
+    from trio._tools.gen_exports import (
+        File,
+        create_passthrough_args,
+        get_public_methods,
+        process,
+        run_linters,
+    )
+except ImportError as error:
+    assert error.msg == "No module named 'astor'"
```

Comment:
Instead of testing an error message, it would be more robust to just import `astor` here, skip on `ImportError`, then do the import of `gen_exports` to run after. You might want to import `isort` and `black` too, to check those are installed.

Reply:
I was thinking something like this would be easier to maintain, in case `gen_exports` is changed not to import `astor` and somebody forgets to update imports in `test_gen_exports`. But you're right that `isort` and `black` can also raise import errors.

Reply:
eh, it got messy when `black` isn't even imported in `gen_exports` - so went with the simpler implementation + added a `noqa`d import for black in `gen_exports` + added a comment to make sure the list in `test_gen_exports` is kept up to date.